### ISSUE TITLE
Build an IconButton that is wrapped in an accessible TooltipHost by default

### ISSUE THREAD

**Issue by mltejera** [microsoft/fluentui#12327]
#### Describe the feature that you would like added
We use IconButtons all over the place in our apps and shared libraries. We're getting hammered by WCAG 2.1 accessibility bugs because the current IconButton doesn't provide a tool tip that explains the intent of the button on keyboard focus. 

We're having to wrap all our icon buttons in a tool tip host, or create a shared component to do this. However it feels like something that we should get from the Fabric component by default.

#### What component or utility would this be added to

IconButton

#### Have you discussed this feature with our team, and if so, who

Not yet - would love an opportunity to, and maybe share back the fix.

**Comment by aneeshack4** [comment]
I believe we've addressed related things briefly in the past - see #11298 - our existing guidance is to just use the Fabric Tooltip as a wrapper. Let me check with the team so if this has changed or if we're willing to take this feature request into consideration for future plans of IconButton.

**Comment by mltejera** [comment]
Agreed - that's exactly the problem we're running into. We working on an internally shared library, so looking for ways to ease our partners pain to meet their WCAG 2.1 requirements.

So far it seems the best way is to create an ToolTipWrappedIconButton that extends button props and has an ITooltipHostProps. Figure that'll make it super easy for consumers to transition and then just add tooltipHost info they need.

**Comment by aneeshack4** [comment]
@mltejera The team discussed this - we do agree that IconButton should have this behavior by default. At present, we want to avoid increasing bundle size and introducing another pattern that will have to be applied to all variants of Button (since IconButton is one of many such variants). I just sent a PR out #12337 that has added an example showing how to wrap `IconButton` with a `Tooltip` and added information under this example about how we'll address providing this behavior out of the box in Fluent as we build the next version of our component library. :D

**Comment by mltejera** [comment]
Wonderful - thank you! 

Is there a task/issue that we can follow tracking the new button work?

**Comment by aneeshack4** [comment]
@mltejera We've started announcing our plans around evolving Fabric into Fluent: https://developer.microsoft.com/en-us/office/blogs/ui-fabric-is-evolving-into-fluent-ui/ We'll follow up there/on the current Fabric site with further details and a roadmap for the components. Everything with the new Fluent component specs is being done in the open though - which are currently [here](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/specs). When we start the IconButton spec, it'll be put there too.
